contracts/pools/weighted-oracle/WeightedOracleMath.sol

```diff
@@ -14,60 +14,88 @@
 
 pragma solidity ^0.7.0;
 
+import "../../lib/math//LogExpMath.sol";
 import "../../lib/math/FixedPoint.sol";
-import "../../lib/math/SignedFixedPoint.sol";
 import "../../lib/math/Math.sol";
 import "../../lib/helpers/InputHelpers.sol";
-import "../../lib/openzeppelin/SafeCast.sol";
 
 /* solhint-disable private-vars-leading-underscore */
 
 contract WeightedOracleMath {
-    using SafeCast for uint256;
-    using SafeCast for int256;
     using FixedPoint for uint256;
-    using SignedFixedPoint for int256;
 
-    int256 internal constant _INVARIANT_COMPRESSION_FACTOR = 1e15;
-    int256 internal constant _PRICE_COMPRESSION_FACTOR = 1e14;
+    int256 private constant _COMPRESSION_FACTOR = 1e14;
+    int256 private constant _HALF_COMPRESSION_FACTOR = 0.5e14;
 
-    function _calculatelInvariantLn(
-        uint256 normalizedWeightA,
-        uint256 balanceA,
-        uint256 normalizedWeightB,
-        uint256 balanceB
-    ) internal pure returns (int256 invariantLn) {
-        //We can cast weights and balances to int256 becuase they are always lower than the max int256.
-        int256 term1 = int256(normalizedWeightA).mul(SignedFixedPoint.ln(int256(balanceA)));
-        int256 term2 = int256(normalizedWeightB).mul(SignedFixedPoint.ln(int256(balanceB)));
+    /**
+     * @dev Returns the natural logarithm of `value`, dropping most of the decimal places to arrive at a value that,
+     * when passed to `_fromLowResLog`, will have a maximum relative error of ~0.05% compared to `value`.
+     *
+     * Values returned from this function should not be mixed with other fixed-point values (as they have different
+     * number of digits), but can be added or subtracted. Use `_fromLowResLog` to undo this process and return to an
+     * 18 decimal places fixed point value.
+     *
+     * Because so much precision is lost, the logarithmic values can be stored using much fewer bits than the original
+     * value required.
+     */
+    function _toLowResLog(uint256 value) internal pure returns (int256) {
```

Comment:
Nit: I'd move these down.
Also, what do you think about `_lowResolutionLog` or `_lowResLog` directly?